### PR TITLE
(CDAP-16590) Log all errors when a scheduled service failed

### PR DESCRIPTION
**Pull request by chtyim**

### CODE REVIEW

**Review by MEseifan — APPROVED** [review]

File: cdap-common/src/main/java/io/cdap/cdap/common/service/AbstractRetryableScheduledService.java [hunk 1]
```diff
@@ -160,15 +160,19 @@ protected final void runOneIteration() throws Exception {
...
         failureCount = 0;
-      } catch (Exception e) {
-        logTaskFailure(e);
+      } catch (Throwable t) {
+        logTaskFailure(t);
+        if (!(t instanceof Exception)) {
+          throw t;
+        }
+        Exception e = (Exception) t;
         if (!shouldRetry(e)) {
```
Comment (CuriousVini) [inline cdap-common/src/main/java/io/cdap/cdap/common/service/AbstractRetryableScheduledService.java:169]: we can use `t` here as well

File: cdap-common/src/main/java/io/cdap/cdap/common/service/AbstractRetryableScheduledService.java [hunk 2]
```diff
@@ -178,9 +182,9 @@ protected final void runOneIteration() throws Exception {
         }
         this.delayMillis = delayMillis;
       }
-    } catch (Exception e) {
-      LOG.error("Aborting service {} due to non-retryable error", getServiceName(), e);
-      throw e;
+    } catch (Throwable t) {
+      LOG.error("Aborting service {} due to non-retryable error", getServiceName(), t);
```
Comment (CuriousVini) [inline cdap-common/src/main/java/io/cdap/cdap/common/service/AbstractRetryableScheduledService.java:186]: usually we do not log and throw - https://github.com/cdapio/hydrator-plugins/pull/1072#discussion_r406868886

File: cdap-common/src/main/java/io/cdap/cdap/common/service/AbstractRetryableScheduledService.java (hunk 1 above)
Comment (chtyim) [reply cdap-common/src/main/java/io/cdap/cdap/common/service/AbstractRetryableScheduledService.java:169]: Can't. The `shouldRetry` only take `Exception`. Technically all "Error" are not retryable.

File: cdap-common/src/main/java/io/cdap/cdap/common/service/AbstractRetryableScheduledService.java (hunk 2 above)
Comment (chtyim) [reply cdap-common/src/main/java/io/cdap/cdap/common/service/AbstractRetryableScheduledService.java:186]: This is what this PR is about. When an `Error` is thrown up in this case, the exception is not getting log at all and the service just terminate. The guava service only log `Exception`.

**Review by CuriousVini — APPROVED** [review]

**Comment by chtyim** [comment]
Build https://builds.cask.co/browse/CDAP-DUT7200

**Pull request merged**